src/settings.cpp

```diff
@@ -61,6 +54,17 @@ Settings::Settings(bool verboseOutput)
     mUpdateOnStartup = false;
     mUnreadOpacityLevel = 0.75;
     mNewEmailMenuEnabled = false;
+
+#ifdef Q_OS_WIN
+    mThunderbirdCmdLine << R"("%ProgramFiles(x86)%\Mozilla Thunderbird\thunderbird.exe")";
+#else
+    if (QFileInfo("/usr/bin/thunderbird").isExecutable()) {
```

Comment:
@ThanosApostolou If Birdtray is started via flatpak, are we still able to start Thunderbird if it was not installed via flatpak? Or will it also not find its libraries and data? If so, we need a way to detect if we are installed via flatpak. Do you know best way to detect that?

Reply:
For birdtray's sandboxed environment, /usr/bin/thunderbird will not be existent. So again we need flatpak-spawn. So the commands are:

  - For flatpak thunderbird
```
flatpak-spawn --host flatpak run org.mozilla.Thunderbird
```
or if you want the exported executable (which does exactly the same)
```
flatpak-spawn --host /var/lib/flatpak/exports/bin/org.mozilla.Thunderbird
```
 - For native thunderbird
```
flatpak-spawn --host thunderbird
```
or if you want full path
```
flatpak-spawn --host /usr/bin/thunderbird
```
 - For snap thunderbird
```
flatpak-spawn --host snap run thunderbird
```


In bash it would be easy to check if a file is existent with
```
flatpak-spawn --host file /usr/bin/thunderbird
```
or even better to find executable thunderbird in any directory in the PATH
```
flatpak-spawn --host sh -c "command -v thunderbird"
```
(`command` is built-in available in shell, that's why we need `sh -c` if we don't want to depend on external tools like `which`).
However I'm not sure how to do this in c++.

Reply:
Well, I certainly don't want to have this kind of complicated logic in code. Can you just provide preconfigured settings, and put the executable command there?

Reply:
@gyunaev I agree that these shouldn't be considered in code. What I would like, if it's easily done, is that the execution of thunderbird can be a command with spaces or arguments and not just a binary path. And then just a simple cmake option which would default to `/usr/bin/thunderbird` and I could set it to `flatpak-spawn --host flatpak run org.mozilla.Thunderbird` during build time. Then I could drop the ugly patch and `/app/bin/thunderbird` script that I use in flathub and users could also easily change the command to something else (which now is hard to do and not documented, since they would need to create a separate script file and override flatpak birdtray to have access to that location).

Reply:
I also agree that there should be no complicated logic in the code, but I'm trying to avoid the cmake configuration parameter at compile time, if we can detect it at run time.
If Birdtray is not able to see the Thunderbird executable when installed via flatpak, then the check above should be suficcient to automatically fall back to the `flatpak-spawn` command without requiring a compile time configuration parameter.

Reply:
@Abestanis  Ok I tested your branch and it seems to work fine with both flatpak and native birdtray. The only problem I saw was that if I delete all the commands/arguments for launching thunderbird from the settings and birdtray is configured to autostart thunderbird, then birdtray refuses to launch (it closes immediately). I think a good approach would be to detect thunderbird command again, if the list is empty. It would be also nice if you can add a button **reset/clear/autodetect** or whatever which would detect thunderbird command from the start, since this change will break existing flatpak installations when I remove my custom **/app/bin/thunderbird** script.

Reply:
@ThanosApostolou Thank you for testing.

> ... then birdtray refuses to launch (it closes immediately)

That was actually a bug that caused Birdtray to crash, good catch. We really need some tests...

> I think a good approach would be to detect thunderbird command again, if the list is empty.

I agree with that, if the user saves an empty path we should just set our default path again. Do you agree with this, @gyunaev? This would be a change from the current behaviour. Currently, we allow an empty executable path and just display an error dialog when we fail to launch Thunderbird.

> add a button reset/clear/autodetect

That is also a good idea, we could even run some more extensive checks here.